### PR TITLE
Notifying Slack for main CI failures.

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -1171,7 +1171,7 @@ jobs:
     
     # Skipped tasks: benchmark, memory, datasize, binsize
     needs: [check-msrv, test, testdata, test-docs, full-datagen, ffi, verify-ffi, gn, features, wasm, fmt, tidy, clippy, doc, ]
-    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
+    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || github.event_name == 'push') }}
```
Comment (Manishearth) [inline .github/workflows/build-test.yml:1174]: issue: for the last one we should `&& branch == main`

Comment (robertbastian) [reply .github/workflows/build-test.yml:1174]: This workflow only runs on push if the branch is main.

Comment (Manishearth) [reply .github/workflows/build-test.yml:1174]: Yes, but I would like to have some redundancy on that

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -1171,7 +1171,7 @@ jobs:
     
     # Skipped tasks: benchmark, memory, datasize, binsize
     needs: [check-msrv, test, testdata, test-docs, full-datagen, ffi, verify-ffi, gn, features, wasm, fmt, tidy, clippy, doc, ]
-    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || github.event_name == 'push') }}
+    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || github.event_name == 'push' && github.ref_name == 'main') }}
```
Comment (Manishearth) [inline .github/workflows/build-test.yml:1174]: nit: please use grouping parentheses on the last one

**Review by Manishearth — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**